chapters/en/slides/module5/module5_04.Rmd

```diff
@@ -0,0 +1,348 @@
+
+If the data is neatly organized in well-separated clusters,
+it is possible that you could visualize more than five colors
+(maybe even ten) effectively,
+but in data where the datapoints are more mixed,
+you will rarely, if ever, be able to go this high.
+
+---
+
+## Too many hues are impossible to distinguish
+
+```{python}
+alt.Chart(cars).mark_point(size=70, filled=True).encode(
+    x='Horsepower',
+    y='Miles_per_Gallon',
+    color=alt.Color('Name', title=None))
+```
+
+Notes:
+This is an example of what happens when too many categorical hues are used.
+In this plot,
+we can not separate the colors
+even if we spent a considerable amount of effort studying the chart.
+
+As we saw on the last slide,
+categorical color schemes have a limited amount of hues
+so in addition of it being hard to differentiate this many hues in general,
+it become practically impossible in cases
+where there are more categories than coor hues,
+since the color scheme starts repeating as in the chart.
+
+A better approach here would have been to label specific points of interest directly
+and keeping the rest as either a single color or using a categorical variable with fewer values, e.g. by grouping the cars into brands rather than their full model name.
+
+Since Altair allows for interactive elements,
+we could also have used the tooltip here as we saw in a previous module.
+
+---
+
+## Redundant coding of bar charts can be disorienting
+
+```{python}
+cars['Brand'] = cars['Name'].str.split().str[0]  # Extract brand from name column
+chart = alt.Chart(cars, width=200, height=450).mark_bar().encode(
+    alt.Y('Brand', title=None, sort='x'),
+    alt.X('mean(Horsepower)'))
+alt.hconcat(chart, chart.encode(alt.Color('Brand', legend=None, scale=alt.Scale(scheme='tableau20'))))
+```
+
+Notes:
+A few slides ago we saw how encoding the same variable as both shape and color
+could make our figures more effective.
+
+Quite frequently,
+you might come across bar charts where the author of the visualization
+has added a distinct color for each bar.
+
+This can work when there are relatively few bars,
+but it usually does not add anything
+and can become directly disorienting when there are many bars
+as in this slide.
+
+It is usually preferable to color all the bars in the same color
+(or after another categorical variable, such as "Origin" in this data),
+and let the axis labels alone identify which category the bar belongs too.
+
+One exception to this is if each axis label
+is broken down into subcategories
+where each location has multiple bars based on another categorical variable,
+then color can help.
+
+---
+
+## Use consistent coloring between subplots even when it is redundant
+
+```{python}
+chart = alt.Chart(cars).mark_bar().encode(
+    alt.X('Origin', title=None),
+    alt.Y('mean(Horsepower)'),
+    alt.Color('Origin'))
+alt.hconcat(chart, chart.mark_line().encode(x='Year', strokeDash='Origin'))
+```
```

Comment:
The second legend here has no colour, can we fix that? Also, do we want origin twice here?

Reply:
There doesn't seem to be an easy why to combine with multiple charts, for a single chart it works as desired.